pages/GoogleMap.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { useEffect, useState } from "react";
+import { GoogleMap, Marker, useJsApiLoader } from "@react-google-maps/api";
+
+interface IMarker {
+  address: string;
+  latitude: number;
+  longitude: number;
+}
+const containerStyle = {
+  width: "1550px",
+  height: "500px",
+};
+
+const center = {
+  lat: 46.7554536,
+  lng: 23.5671444,
+};
+
+const marker1 = {
+  lat: 46.7554536,
+  lng: 23.5671444,
+};
+
+const marker2 = {
+  lat: 46.7546919,
+  lng: 23.5604183,
+};
+
+type GoogleLatLng = google.maps.LatLng;
+type GoogleMarker = google.maps.Marker;
+
+function MyComponent() {
+  const { isLoaded } = useJsApiLoader({
+    id: "google-map-script",
+    googleMapsApiKey: process.env.NEXT_PUBLIC_APP_GOOGLE_MAPS_API_KEY as any,
+  });
+
+  const [marker, setMarker] = useState<IMarker>();
+
+  const [map, setMap] = useState<GoogleMap>();
+
+  const onLoad = React.useCallback(function callback(map) {
+    const bounds = new window.google.maps.LatLngBounds();
+    map.fitBounds(bounds);
+    setMap(map);
+  }, []);
+
+  const initEventListener = (): void => {
+    if (map) {
+      google.maps.event.addListener(map, "click", function (e: any) {
+        coordinateToAddress(e.latLng);
```

Comment:
Please move the contents of the method directly here, avoid defining methods outside react hooks.